### ISSUE TITLE
False positive for weak_delegate rule since Xcode 9.3

### ISSUE THREAD

**Issue by sroebert** [realm/SwiftLint#2150]
### New Issue Checklist

- [ ] Updated SwiftLint to the latest version
- [ ] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Bug Report

False positive for `weak_delegate` rule since Xcode 9.3.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint --path file.swift
Loading configuration from '.swiftlint.yml'
Linting Swift files at path file.swift
Linting 'file.swift' (1/1)
file.swift:1:1: warning: Weak Delegate Violation: Delegates should be weak to avoid reference cycles. (weak_delegate)
Done linting! Found 1 violation, 0 serious in 1 file.
```

```swift
protocol SomeDelegate: class {
}
class SomeClass {
    weak var delegate: SomeDelegate?
}
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? **0.25.1**
* Installation method used (Homebrew, CocoaPods, building from source, etc)? **Homebrew**
* Paste your configuration file: **Not needed**

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents. **No**
* Which Xcode version are you using (check `xcode-select -p`)? **9.3**

**Comment by marcelofabri** [comment]
Are you sure this sample example triggers and that you're using 0.25.1?

I wasn't able to reproduce:

```
$ swiftlint --path file.swift
Linting Swift files at path file.swift
Linting 'file.swift' (1/1)
Done linting! Found 0 violations, 0 serious in 1 file.
$cat file.swift 
protocol SomeDelegate: class {
}
class SomeClass {
    weak var delegate: SomeDelegate?
}
```
